cpp/benchmarks/stream_compaction/unique.cpp

```diff
@@ -62,9 +66,12 @@ void nvbench_unique(nvbench::state& state, nvbench::type_list<Type, nvbench::enu
   auto input_column = source_column->view();
   auto input_table  = cudf::table_view({input_column, input_column, input_column, input_column});
 
+  auto const sort_order = cudf::sorted_order(input_table);
+  auto const sort_table = cudf::gather(input_table, *sort_order);
+
   state.set_cuda_stream(nvbench::make_cuda_stream_view(cudf::get_default_stream().value()));
   state.exec(nvbench::exec_tag::sync, [&](nvbench::launch& launch) {
-    auto result = cudf::unique(input_table, {0}, Keep, cudf::null_equality::EQUAL);
+    auto result = cudf::unique(*sort_table, {0}, Keep, cudf::null_equality::EQUAL);
```

Comment:
I think we can do better by benchmarking both with and w/o sorting input. This is is what I was expecting, and we can achieve that very easily.
```
auto const run_bench = [&] (auto const input){
  state.set_cuda_stream();
  state.exec(...);
};
auto const sorting = static_cast<bool>(state.get_int64("Sort"));
if(sorting) {
   // do the benchmark with sorted table
   run_bench(sort_table);
} else {
   // do the benchmark with the random generated input table
   run_bench(rnd_table);
}
```